### ISSUE TITLE
Explore Accessibility and labeling Options for AILabel Component

### ISSUE THREAD

**Issue by 2nikhiltom** [carbon-design-system/carbon#18241]
Originated in this [slack thread  ](https://ibm-studios.slack.com/archives/C036P1CTN/p1733731510915439) while discussing about `Accessible name does not match or contain the visible label text` violation in [AILabel](https://react.carbondesignsystem.com/?path=/story/components-ailabel--default) (false pisitive)

Redundant ARIA labeling concern 
The redundancy concern stems from the fact that the visible text ("AI Text goes here") within the button is already accessible and will be announced by screen readers as the button's accessible name. Specifying the same text again using the aria-label attribute creates duplication

Proposed Solution the thread discussion:
Refine the interaction and improve accessibility by introducing `tooltips`
- Display a `Tooltip` on hover/focus of the button to provide additional information. This would make the content more discoverable and enhance the accessibility for users who rely on hover or focus events.
- And still Maintain the existing `Toggletip` behavior for displaying more detailed information when the button is clicked or focused, ensuring that the button's functionality remains interactive and provides the necessary explainability when triggered.

Update label to `AI explanation` / `AI details` / `AI info` since`AI` as a button name is pretty scant

- [ ] Explore more potential ideas/workarounds
